js/packages/openinference-instrumentation-llama-index/test/llamaIndex.test.ts

```diff
@@ -320,3 +338,136 @@ describe("LlamaIndexInstrumentation", () => {
+  });
+
+  it("is patched", () => {
+    expect(isPatched()).toBe(true);
+  });
+
+  it("isLLMPrototype should identify LLM prototypes correctly", async () => {
+    // Expect all retriever prototypes to be identified as a retriever
+    expect(isLLMPrototype(llamaindex.OpenAI.prototype)).toEqual(true);
+    expect(isLLMPrototype(llamaindex.MistralAI.prototype)).toEqual(true);
+    expect(isLLMPrototype(llamaindex.Gemini.prototype)).toEqual(true);
+    expect(isLLMPrototype(llamaindex.HuggingFaceLLM.prototype)).toEqual(true);
+
+    // Expect a non-retriever object to be identified as such
+    expect(isLLMPrototype({})).toEqual(false);
+    expect(isLLMPrototype(null)).toEqual(false);
+    expect(isLLMPrototype(undefined)).toEqual(false);
+    expect(isLLMPrototype(RetrieverQueryEngine.prototype)).toEqual(false);
+    expect(isLLMPrototype(HuggingFaceEmbedding.prototype)).toEqual(false);
+    expect(isLLMPrototype(llamaindex.TextNode.prototype)).toEqual(false);
+    expect(isLLMPrototype(llamaindex.CorrectnessEvaluator.prototype)).toEqual(
+      false,
+    );
+  });
+
+  it("should create a span for LLM chat - Query Engine", async () => {
+    // Create Document object with essay
+    const document = new Document({ text: "lorem ipsum" });
+
+    // Split text and create embeddings. Store them in a VectorStoreIndex
+    const index = await VectorStoreIndex.fromDocuments([document]);
+
+    // Query the index
+    const queryEngine = index.asQueryEngine();
+    const response = await queryEngine.query({
+      query: "What did the author do in college?",
+    });
+
+    // Output response
+    expect(response.response).toEqual(RESPONSE_MESSAGE.content);
+
+    const spans = memoryExporter.getFinishedSpans();
+    expect(spans.length).toBeGreaterThan(0);
+
+    // Expect a span for the query engine
+    const LLMSpan = spans.find((span) => span.name.includes("llm"));
+    expect(LLMSpan).toBeDefined();
+
+    // Verify span attributes
+    expect(
+      LLMSpan?.attributes[SemanticConventions.OPENINFERENCE_SPAN_KIND],
+    ).toEqual(OpenInferenceSpanKind.LLM);
+
+    expect(LLMSpan?.attributes[SemanticConventions.LLM_MODEL_NAME]).toEqual(
+      CHAT_RESPONSE.model,
+    );
+
+    expect(
+      LLMSpan?.attributes[SemanticConventions.LLM_INVOCATION_PARAMETERS],
+    ).toEqual(
+      '{"model":"gpt-3.5-turbo","temperature":0.1,"topP":1,"contextWindow":4096,"tokenizer":"cl100k_base"}',
+    );
+
+    expect(
+      LLMSpan?.attributes[
+        `${SemanticConventions.LLM_INPUT_MESSAGES}.0.${SemanticConventions.MESSAGE_ROLE}`
+      ],
+    ).toEqual("user");
+    expect(
+      LLMSpan?.attributes[
+        `${SemanticConventions.LLM_INPUT_MESSAGES}.0.${SemanticConventions.MESSAGE_CONTENT}`
+      ],
+    ).toEqual(
+      "Context information is below.\n" +
+        "---------------------\n" +
+        "lorem ipsum\n" +
+        "---------------------\n" +
+        "Given the context information and not prior knowledge, answer the query.\n" +
```

Comment:
for this portion of the query where does it come from? is this just injected from llama index? did you plug in a prompt template somewhere here? I see the document content and the query below bu curious about the rest